esp/templates/program/modules/onsitecheckinmodule/barcodecheckin.html

```diff
@@ -57,17 +112,133 @@ <h1>Barcode Checkin &mdash; For {{ program.niceName }}</h1>
 
 <form id="checkinform" name="checkinform" method="POST" action="{{ request.path }}">
                     {{ form.as_p }}
-                    <p>
+                    <p class="ids">
                 <input type="submit" value="Submit" />
                 </p>
 </form>
 
-<br />
+<br><br>
+
+<div class="controls">
+    <button type="button" class="scan"><i class="fas fa-barcode"></i>&ensp;Scan barcodes with your device</button>
+    &emsp;
+    <select>
+        <option selected value="code_39_reader">Code 39</option>
+        <option value="codabar_reader">Codabar</option>
+        <option value="code_128_reader">Code 128</option>
+        <option value="i2of5_reader">Interleaved 2 of 5</option>
+    </select>
+</div>
+<div class="overlay overlay--inline">
+    <div class="overlay__content viewport" id="interactive">
+        <div class="overlay__close">X</div>
+    </div>
+</div>
+
+<!--<br />
 {% load render_qsd %}
-{% render_inline_program_qsd program "onsite:status" %}
+{% render_inline_program_qsd program "onsite:status" %}-->
 
 
 </div>
 
+<script type="text/javascript">
+$j(function() {
+    var App = {
+        init: function() {
+            this.attachListeners();
+        },
+        attachListeners: function() {
+            var button = document.querySelector('button.scan'),
+                self = this;
 
-{% endblock %}
+            button.addEventListener("click", function clickListener (e) {
+                e.preventDefault();
+                button.removeEventListener("click", clickListener);
+                self.activateScanner();
+            });
+        },
+        activateScanner: function() {
+            var scanner = this.configureScanner('.overlay__content'),
+                onDetected = function (result) {
+                    this.addToResults(result);
+                }.bind(this),
+                stop = function() {
+                    Quagga.stop();
+                    console.log("Scanner stopped")
```

Comment:
Should we get rid of all the console.log lines? Normally, I would say yes, but since this is functionality only used by admins, I could see an argument for keeping them.

Reply:
This was mostly for debugging, but I could also seeing it be valuable to admins in case they are having trouble. However, given that most use of this will be on mobile devices, and you can't see console logs on mobile, it probably isn't very helpful after all?